source/slang/type-layout.h

```diff
@@ -358,6 +358,13 @@ class StructTypeLayout : public TypeLayout
     Dictionary<Decl*, RefPtr<VarLayout>> mapVarToLayout;
 };
 
+class GenericParamTypeLayout : public TypeLayout
+{
+public:
+    RefPtr<GlobalGenericParamDecl> decl;
```

Comment:
I know this is silly, but: you should haven't to store this, because the `type` field of the `TypeLayout` in this case should always be a `DeclRefType` which references the `GlobalGenericParamDecl`, so this field is redundant (and we could just have a convenience accessor instead).